src/App.tsx

```diff
@@ -15,36 +15,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'Alphabet',
+  Length = 'Length',
+  Default = '',
+}
+
+function prepareGoods(goods: string[], type: SortType, reverse: boolean) {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (type) {
+      case SortType.Alphabet:
+        return good1.localeCompare(good2);
+      case SortType.Length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortQuery, setSortQuery] = useState<SortType>(SortType.Default);
+  const [reversed, setReversed] = useState(false);
+
+  const sortedGoods = prepareGoods(goodsFromServer, sortQuery, reversed);
+
+  const isChanged = sortQuery || reversed;
+
+  const resetFilter = () => {
+    setSortQuery(SortType.Default);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
```

Comment:
Perhaps, it will be better to move this logic to separate component `<Button `

```
const Button = ({
  active,
  onClick,
  children,
}:
```

```
        <Button active={sortType === SortType.Alphabet} onClick={() => setSortType(SortType.Alphabet)}>
          Sort alphabetically
        </Button>
```